lib/Sema/CSSimplify.cpp

```diff
@@ -1473,6 +1474,15 @@ shouldOpenExistentialCallArgument(ValueDecl *callee, unsigned paramIdx,
   return result;
 }
 
+static bool isDirectConformanceOf(TypeVariableType *typeVar,
+                                  Constraint *conformance) {
+  ASSERT(conformance->getKind() == ConstraintKind::ConformsTo ||
+         conformance->getKind() == ConstraintKind::NonisolatedConformsTo);
```

Comment:
Given both callers check this before calling, would it make more sense for it to be conditional here instead?